core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/UdtCodec.java

```diff
@@ -105,10 +105,7 @@ public UdtValue decode(@Nullable ByteBuffer bytes, @NonNull ProtocolVersion prot
       int i = 0;
       while (input.hasRemaining()) {
         if (i == cqlType.getFieldTypes().size()) {
-          throw new IllegalArgumentException(
-              String.format(
-                  "Too many fields in encoded UDT value, expected %d",
-                  cqlType.getFieldTypes().size()));
+          break;
```

Comment:
I can understand not failing in this case, but logging a (no-spam) warning seems appropriate. Based on my understanding, this should only happen if a schema change happens on the server to add fields to a UDT, and the client gets an updated UDT in a response before finding out about the new schema.

Are there any other situations when this might occur?

Reply:
Yes, that's the scenario we ran into, and I can't think of any other scenarios.
Fair, a no-spam warn is a good idea but it doesn't look like the client currently has a no-spam logger. I could copy the one from the server but maybe not the best thing to do in this PR... and a vanilla log warn would be bad here since it would get logged every time we decode a UDT field matching that scenario. Thoughts?

Reply:
Thought about this some more. The driver seeing extra fields in a UDT is expected for a short period after a schema change to add fields, since the control connection may find out later than the coordinator or replicas, and EVENTs are sent over the control connection async. It may be worth a warning if this state persists for too long, since that's an indication that a schema change EVENT wasn't delivered, but delivery failures are expected to be tolerated as well.

I think it makes sense to drop this to a DEBUG or TRACE log, rather than using a no-spam logger.

In the more general case of EVENT delivery failing and client metadata going out of sync, there's more we could do to retry EVENTs or support bounded metadata lag, but all of that is out of scope for this ticket.

The only other change I'm considering is whether we should add a hint to refresh client metadata in this situation where the server has indicated that it has a new schema the client isn't yet aware of. Metadata refreshes should be debounced so we don't attempt a refresh on every response decode. Refreshing client metadata would only help in the case when the control instance does know about the schema but EVENT delivery failed, not in cases where the control instance has not yet found out about the new schema but a particular coordinator has, which is more likely the case for larger clusters.

Reply:
It's been a while but my memory is a little fuzzy as to the exact scenario with which we ran into this, but it had to do with the object mapper caching schemas somewhere too. In any case, the client metadata refresh hint sounds reasonable for when we tackle what you said earlier about EVENT delivery failing, since that would be the logical place for the hint to propagate and affect. For now I'll add a debug log.

Reply:
I think if we had an existing `NoSpamLogger`, logging this once at `WARN` would be pretty informative;  Maybe something we can do as a follow on PR later?  Wondering if there's anything else in the driver that could benefit from a no spam logger...

Reply:
Spent a while paging in the context for this tonight, another piece of it is that the old schema will usually get cached  by the CachingCodecRegistry and you can't overwrite it because the driver thinks it collides, so the driver will have the old UdtCodec until it gets bounced/reinitialized.